src/engine/source/base/src/utils/stringUtils.cpp

```diff
@@ -204,16 +204,22 @@ bool isNumber(const std::string& str)
 
 bool replaceAll(std::string& data, const std::string_view toSearch, const std::string_view toReplace)
 {
-    auto pos {data.find(toSearch)};
-    const auto ret {std::string::npos != pos};
+    if (toSearch.empty() || toSearch == toReplace)
```

Comment:
```suggestion
    if (toSearch.empty() || toSearch == toReplace || toReplace.find(toSearch) != std::string_view::npos)
```

Reply:
to avoid this issue:

add test with 

```
    std::string input = "a";
    replaceAll(input, "a", "aa");
```

Reply:
Indeed it does
![image](https://github.com/user-attachments/assets/d8224a44-9d43-4332-b089-a262d66e82b1)

Reply:
Thank you both! Good catch!